src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+      );
+
+      useFloatingSpy.mockRestore();
+    });
+
+    it("renders a validation icon on the label if validationOnLabel passed as true", () => {
+      render(
+        <Textarea
+          {...{
+            label: "Label",
+            [validationProp]: "Message",
+            validationOnLabel: true,
+          }}
+        />
+      );
+      const labelContainer = screen.getByTestId("label-container");
+      const validationIcon = screen.getByTestId(`icon-${validationProp}`);
+      expect(labelContainer).toContainElement(validationIcon);
+    });
+
+    it("overrides the tooltip position on the validation icon when the tooltipPosition prop is set and validationOnLabel is true", () => {
+      const useFloatingSpy = jest.spyOn(floatingUi, "useFloating");
+
+      render(
+        <Textarea
+          {...{
+            label: "Label",
+            [validationProp]: "Message",
+            tooltipPosition: "bottom",
+            validationOnLabel: true,
+          }}
+        />
+      );
+
+      expect(useFloatingSpy).toHaveBeenCalledWith(
+        expect.objectContaining({ placement: "bottom" })
+      );
+
+      useFloatingSpy.mockRestore();
+    });
+  }
+);
+
+test.each(validationTypes)(
+  "when %s validation prop is true boolean, does not render any validation iconn",
+  (validationProp) => {
+    render(<Textarea {...{ [validationProp]: true }} />);
+    expect(
+      screen.queryByTestId(`icon-${validationProp}`)
+    ).not.toBeInTheDocument();
+  }
+);
+
+test("should set the aria-label on the Help component to the value of the helpAriaLabel prop", () => {
+  render(<Textarea label="foo" labelHelp="bar" helpAriaLabel="baz" />);
+
+  expect(screen.getByRole("button")).toHaveAttribute("aria-label", "baz");
+});
+
+test("should set the Help component's text content to the value of the labelHelp prop", () => {
+  render(<Textarea label="foo" labelHelp="bar" helpAriaLabel="baz" />);
+
+  screen.getByRole("button").focus();
+  expect(screen.getByTestId("tooltip")).toHaveTextContent("bar");
+});
+
+test.each(["info", "warning", "error"])(
+  "with id prop provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(<Textarea id="foo" label="bar" {...{ [validationType]: "test" }} />);
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(textarea).toHaveAttribute("aria-describedby", "foo-validation");
+  }
+);
+
+test.each(["info", "warning", "error"])(
+  "with id prop not provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(<Textarea label="bar" {...{ [validationType]: "test" }} />);
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(textarea).toHaveAttribute(
```

Comment:
suggestion: you could use [toHaveAccessibleDescription](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohaveaccessibledescription) here